assets/color-swatch.css

```diff
@@ -0,0 +1,18 @@
+.color-swatch-name {
+  pointer-events: none;
+  transition: transform 0.2s;
+}
+
+.product-form__input--color-swatch input.input-color-swatch+.color-swatch:hover .color-swatch-name {
+  transform: translate(0, -30px);
+}
+
+.product-form__input--color-swatch input.input-color-swatch:checked+.color-swatch {
+  color: rgba(18, 18, 18, 0.75);
```

Comment:
Always use variables for color properties, since this is vanila CSS, you can use the CSS variables.